### PR TITLE
Refactor WashingtonPost Handlers

### PR DESCRIPTION
**Pull request by SebastianSchmidl**
see #16 and #15

### CODE REVIEW

File: cse/pipeline/SynchronousHandlerContext.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-class SynchronousHandlerContext:
+class SynchronousHandlerContext(object):
```
Comment (Benedikt1992) [inline cse/pipeline/SynchronousHandlerContext.py:1]: Why inherit the object class?

File: cse/SearchEngine.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
+class SearchEngine(object):
```
Comment (Benedikt1992) [inline cse/SearchEngine.py]: This shouldn't be part of this PR, should it?

File: cse/pipeline/Pipeline.py [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import abc
 
-class Pipeline:
+class Pipeline(object):
```
Comment (Benedikt1992) [inline cse/pipeline/Pipeline.py:3]: Why inherit the object class?

File: cse/SearchEngine.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
```
Comment (Benedikt1992) [inline cse/SearchEngine.py]: This shouldn't be part of this PR, should it?

File: cse/pipeline/Pipeline.py (hunk 3 above)
Comment (SebastianSchmidl) [reply cse/pipeline/Pipeline.py:3]: > In Python 3, classes extend object implicitly, whether you say so yourself or not.
>
> In Python 2, there's old-style and new-style classes. To signal a class is new-style, you have to inherit explicitly from object. If not, the old-style implementation is used.
see https://docs.python.org/release/2.5.2/ref/node33.html

Added for backwards compatibility to python2. This is needed for this PR, because getting the right class name for the handlers requires new-style classes

Comment (SebastianSchmidl) [reply cse/pipeline/Pipeline.py:3]: I wanted to have consistent class declarations, so I changed all classes to inherit from object (be new-style classes in python2 as well)

**Review by Benedikt1992 — APPROVED** [review]
